narwhals/_duration.py

```diff
@@ -0,0 +1,50 @@
+"""Tools for working with the Polars duration string language."""
+
+from __future__ import annotations
+
+import re
+from typing import TYPE_CHECKING
+from typing import Literal
+from typing import cast
+from typing import get_args
+
+if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
+
+__all__ = ["IntervalUnit", "parse_interval_string"]
+
+IntervalUnit: TypeAlias = Literal["ns", "us", "ms", "s", "m", "h", "d", "mo", "q", "y"]
```

Comment:
Thinking this will be helpful when we're far away from the public `.dt` methods

```suggestion
IntervalUnit: TypeAlias = Literal["ns", "us", "ms", "s", "m", "h", "d", "mo", "q", "y"]
"""A Polars duration string interval unit.

- 'ns': nanosecond.
- 'us': microsecond.
- 'ms': millisecond.
- 's': second.
- 'm': minute.
- 'h': hour.
- 'd': day.
- 'mo': month.
- 'q': quarter.
- 'y': year.
"""
```